### PR TITLE
Refactor test_common.py

### PR DESCRIPTION
**Pull request by JevinJ**
### Overview

Quick refactoring pass in test_common.

In a future PR I can break the tests up more and try to get better coverage with hypothesis, if there's no issue with that. I'm not liking in a lot of places the tests don't say what exactly they're testing, or they test several things at once.

### Details

- Add a grid fixture.
- Fix typos, double statements, unnecessary code
- Add missing asserts
- Fix a test which doesn't get picked up by pytest.

### CODE REVIEW

**Comment by akaszynski** [comment]
Good usage of `fixtures`.  That's actually a feature I only just realized existed and it's down on the list for needing to be implemented.

File: tests/test_common.py [hunk 1]
```diff
@@ -669,7 +618,5 @@ def test_find_closest_point():
     assert len(index) == 5
 
 
-def test_setting_points_from_self():
-    grid = GRID.copy()
-    grid.points = GRID.points
-    assert np.allclose(grid.points, GRID.points)
+def test_setting_points_from_self(grid):
+    assert np.allclose(grid.points, grid.points)
```
[the remark this answers is not in the document]
Comment (JevinJ) [reply tests/test_common.py]: Fixed

**Comment by akaszynski** [comment]
Doing these refactors always scares me.  These are relatively harmless, but if we start introducing false positives into our test matrix, things are going to break down the line.

Refactoring will help with code readability, but where we really need effort is with code coverage so we don't have to keep building the docs to have full coverage.  For example, you could add additional tests in this or another PR that improves the coverage of `common.py`:
https://codecov.io/gh/pyvista/pyvista/src/master/pyvista/core/common.py

We should aim 100% coverage wherever possible.  I also agree about including docs for unit tests.  These should be self documenting, but for cases where they're not, we absolutely should add a comment or two of why we're doing a test.

**Comment by akaszynski** [comment]
As for code coverage, what I find most helpful is running:
```
pytest --cov pyvista --cov-report annotate
```
That way I have a report locally of what's good and what I'm missing and can see if my unit tests are hitting the right lines.

**Comment by akaszynski** [comment]
Please wait to merge this until we've closed #669, and open a new PR with additional coverage if that's an issue you want to tackle.

**Comment by JevinJ** [comment]
> 
> 
> Doing these refactors always scares me. These are relatively harmless, but if we start introducing false positives into our test matrix, things are going to break down the line.
> 
> Refactoring will help with code readability, but where we really need effort is with code coverage so we don't have to keep building the docs to have full coverage. For example, you could add additional tests in this or another PR that improves the coverage of `common.py`:
> https://codecov.io/gh/pyvista/pyvista/src/master/pyvista/core/common.py
> 
> We should aim 100% coverage wherever possible. I also agree about including docs for unit tests. These should be self documenting, but for cases where they're not, we absolutely should add a comment or two of why we're doing a test.

@akaszynski, what I want to do with tests is:

-  Break them up so multiple things are tested at once, and fewer repeat tests. 

- Rename and group them in a style like [DataSetAttributes](https://github.com/pyvista/pyvista/pull/659/files#diff-8b0a87a55f301e5232efe3f4113a30da), so each test is logically grouped, and tells you what it's expecting to happen.

- Use Hypothesis where possible.

- Don't change the logic, just move it, I can do this in small groups/PRs so it's easier to manage.

**Comment by akaszynski** [comment]
Is there a way to get hypothesis to work with pytest?  I really like the approach we have here, but I’m open to a change if it improves our testing framework.

I’m all for refactoring and making the tests more ordered and grouped, and ideally at the same time we’d improve our coverage as well.

**Comment by JevinJ** [comment]
@akaszynski 

I *think* hypothesis pytest support is built in, but there is a package called 'hypothesis-pytest' which adds the functionality.

File: tests/test_common.py [hunk 2]
```diff
@@ -570,9 +523,7 @@ def test_string_arrays():
 def test_clear_arrays():
     # First try on an empty mesh
     grid = pyvista.UniformGrid((10, 10, 10))
-    grid.clear_arrays()
     # Now try something more complicated
-    grid = GRID.copy()
```
Comment (akaszynski) [inline tests/test_common.py:575]: Just noticed this.  Any reason why we change the grid that we're using here?  We start off with a uniform grid and then change it to our hex example above.  Maybe split up the test so we check clearing the array on both types?  Would `hypothesis` be a good case for this?

File: tests/test_common.py [hunk 3]
```diff
@@ -523,16 +478,14 @@ def test_get_cell_array_fail():
...
     assert grid.extent is None
 
 
-
-def set_cell_vectors():
-    grid = GRID.copy()
+def test_set_cell_vectors(grid):
     grid.cell_arrays['_cell_vectors'] = np.random.random((grid.n_cells, 3))
     grid.set_active_vectors('_cell_vectors')
+    assert grid.active_vectors_name == '_cell_vectors'
```
Comment (akaszynski) [inline tests/test_common.py:488]: Thanks, this was missing.

File: tests/test_common.py (hunk 2 above)
Comment (akaszynski) [reply tests/test_common.py:575]: Functionally, the difference is that the `UniformGrid` is already empty, whereas `grid` already contains scalars.  Maybe a separate test that clears it twice?

Comment (akaszynski) [reply tests/test_common.py:575]: We can do this on a separate PR and merge this now.

**Pull request merged**